beacon_node/beacon_chain/src/data_availability_checker/overflow_lru_cache.rs

```diff
@@ -776,6 +776,21 @@ impl<T: BeaconChainTypes> OverflowLRUCache<T> {
             drop(write_lock);
         }
 
+        // Collect keys of pending blocks from a previous epoch to cutoff
+        let mut write_lock = self.critical.write();
+        let mut keys_to_remove = vec![];
+        for (key, value) in write_lock.in_memory.iter() {
+            if let Some(epoch) = value.epoch() {
+                if epoch < cutoff_epoch {
+                    keys_to_remove.push(*key);
+                }
+            }
+        }
+        // Now remove keys
+        for key in keys_to_remove {
+            write_lock.in_memory.pop(&key);
```

Comment:
this just stops the cache from building up but won't clear old junk, will it?

Reply:
There's a second maintenance routine that prunes keys in disk that are not in the memory map